app/src/main/java/com/conupods/OutdoorMaps/View/Settings/SettingsActivity.java

```diff
@@ -24,35 +24,38 @@ protected void onCreate(Bundle savedInstanceState) {
         Button personalPage = findViewById(R.id.toggle2_1);
         Button infoPage = findViewById(R.id.toggle2_2);
         //Preferences buttons
-        CheckBox concordiaShuttle = findViewById(R.id.concordiaShuttle);
-        CheckBox elevators = findViewById(R.id.elevators);
-        CheckBox escalators = findViewById(R.id.escalators);
-        CheckBox stairs = findViewById(R.id.stairs);
-        CheckBox accessibilityInfo = findViewById(R.id.accessibilityInfo);
-        CheckBox stepFreeTrips = findViewById(R.id.stepFreeTrips);
+        CheckBox[] preferenceButtons = {findViewById(R.id.concordiaShuttle), findViewById(R.id.elevators), findViewById(R.id.escalators),
+                findViewById(R.id.stairs), findViewById(R.id.accessibilityInfo), findViewById(R.id.stepFreeTrips)};
         //Top menu events
         done.setOnClickListener(view -> startActivityIfNeeded(new Intent(SettingsActivity.this, MapsActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0));
         personalPage.setOnClickListener(view -> startActivityIfNeeded(new Intent(SettingsActivity.this, SettingsPersonalActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0));
         infoPage.setOnClickListener(view -> startActivityIfNeeded(new Intent(SettingsActivity.this, SettingsInfoActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0));
         //Check Default preferences
         setCheckedBoxes();
         //My preferences events
-        concordiaShuttle.setOnClickListener(view -> changePreferences(concordiaShuttle));
-        elevators.setOnClickListener(view -> changePreferences(elevators));
-        escalators.setOnClickListener(view -> changePreferences(escalators));
-        stairs.setOnClickListener(view -> changePreferences(stairs));
-        accessibilityInfo.setOnClickListener(view -> changePreferences(accessibilityInfo));
-        stepFreeTrips.setOnClickListener(view -> changePreferences(stepFreeTrips));
+        preferenceButtons[0].setOnClickListener(view -> changePreferences(preferenceButtons[0], preferenceButtons));
```

Comment:
use a for loop-each

Reply:
sure i dont know why i didnt think about that